clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -680,7 +680,7 @@ private boolean maybeValidatePosition(Metadata.LeaderAndEpoch currentLeaderAndEp
                 return false;
             }
 
-            if (position != null && !position.safeToFetchFrom(currentLeaderAndEpoch)) {
+            if (position != null && !position.currentLeader.equals(currentLeaderAndEpoch)) {
```

Comment:
+1, the above check for empty leader and epoch obviates the extra check we were doing in safeToFetchFrom

Reply:
Yeah, I was happy to simplify this. I think I was responsible for the initial confusing check.